### PR TITLE
Extend kubectl example with apply + edit and example improvements

### PR DESCRIPTION
**Pull request by clux**
Improves `kubectl` example  - see [this file](https://github.com/kube-rs/kube-rs/pull/897/files#diff-8624a73a259e50f57811c3d74a23d2a5e9cb50c1db51db8a27927864db1ab72f))
- `apply` support via yaml from multidoc files using helpers from #896 + serverside apply dynapi
- `edit` using `edit` crate (reasonably small for examples only)
- cleans up api generation
- factored out behaviour into methods on the clap struct

Plus a general examples cleanup:
- removed `configmap_watcher` (the least advanced one, and plenty of watcher examples already)
- removed `NAMESPACE` evar use as a bad substitute for `Api::default_namespaced` + `kns`

rationale for last one: much better to let people rely on their namespace switchers like `kns` rather than having us implement `NAMESPACE` everywhere..

### CODE REVIEW

File: examples/kubectl.rs [hunk 1]
```diff
@@ -72,102 +74,124 @@ async fn main() -> Result<()> {
...
-            Api::default_namespaced_with(client.clone(), &ar)
+    // specialized handling for apply (can handle multiple resources)
+    if verb == "apply" {
+        let ssapply = PatchParams::apply("kubectl-light").force();
+        if let Some(pth) = file {
+            let yaml =
+                std::fs::read_to_string(&pth).with_context(|| format!("Failed to read {}", pth.display()))?;
+            for doc in multidoc_deserialize(&yaml)? {
+                let yaml_name = doc.get("metadata").map(|m| m.get("name").map(|v| v.as_str()));
+                let tm: TypeMeta = serde_yaml::from_value(doc.clone())?;
```
Comment (clux) [inline examples/kubectl.rs:97]: this clone is slightly annoying :/

Comment (clux) [reply examples/kubectl.rs]: made it smarter by going through DynamicObject

File: examples/kubectl.rs [hunk 2]
```diff
@@ -72,102 +74,140 @@ async fn main() -> Result<()> {
...
+            } else {
+                warn!("need a name to edit");
             }
-        }
-    } else if verb == "delete" {
-        if let Some(n) = &name {
-            if let Either::Left(pdel) = api.delete(n, &Default::default()).await? {
-                // await delete before returning
-                await_condition(api, n, is_deleted(&pdel.uid().unwrap())).await?
+        } else if verb == "get" {
```
Comment (clux) [inline examples/kubectl.rs:142]: everything below here in main is basically unchanged, mostly just indented

File: examples/kubectl.rs [hunk 3]
```diff
@@ -18,19 +18,22 @@ use kube::{
...
     output: OutputMode,
+    #[clap(long, short)]
+    file: Option<std::path::PathBuf>,
     #[clap(long, short = 'l')]
     selector: Option<String>,
     #[clap(long, short)]
     namespace: Option<String>,
     #[clap(long, short = 'A')]
     all: bool,
     verb: String,
```
Comment (kazk) [inline examples/kubectl.rs:35]: Can `verb` be enum as well?

**Review by kazk — APPROVED** [review]

**Pull request merged**